integration-tests/helpers.js

```diff
@@ -38,6 +41,69 @@ class FakeAgent extends EventEmitter {
         payload: msgpack.decode(req.body, { codec })
       })
     })
+    app.post('/v0.7/config', (req, res) => {
+      const buffers = []
+      req.on('data', buffers.push.bind(buffers))
+      req.on('end', () => {
+        const { products } = JSON.parse(Buffer.concat(buffers).toString()).client
+
+        const expires = (new Date(Date.now() + 1000 * 60 * 60 * 24)).toISOString() // in 24 hours
+        const clientID = uuid() // TODO: What is this? It isn't the runtime-id
+
+        // Currently, only `opaque_backend_state` and `targets` are used by dd-trace-js in the object below
+        const targets = {
+          signatures: [],
+          signed: {
+            _type: 'targets',
+            custom: {
+              agent_refresh_interval: 5,
+              opaque_backend_state: ''
+            },
+            expires,
+            spec_version: '1.0.0',
+            targets: {},
+            version: 12345
+          }
+        }
+        const opaqueBackendState = {
+          version: 2,
+          state: { file_hashes: { key: [] } }
+        }
+        const targetFiles = []
+        const clientConfigs = []
+
+        const files = this._rc_files.filter(({ product }) => products.includes(product))
+
+        for (const { orgId, product, id, name, config } of files) {
```

Comment:
when a config file is already received and applied by the library, it's specified in the `cached_target_files` field, and the agent is supposed to not send them again to optimize it. I think sending them should not cause any problem as we have a check that make sure to ignore config files that have the same hashes as already applied ones, but I'm mentioning it just in case.

Reply:
Fixed 👍